pkg/rancher-desktop/components/MarketplaceCatalog.vue

```diff
@@ -100,8 +108,9 @@ export default (Vue as VueConstructor<Vue & VuexBindings>).extend({
       >
         <MarketplaceCard
           :extension="item"
+          :revalidate-state="isInstalled"
```

Comment:
I think the component design can be simplified by emitting an event instead of passing a function, for example 

```suggestion
          @update:extension="isInstalled"
```

Following this pattern would allow us to remove the `revalidateState` prop entirely and shift the responsibility of checking if an extension is installed solely on the caller.

(instead of invoking `this.revalidateState();` in `MarketplaceCard.vue`, we can utilize something along the lines of `this.$emit('update:extension')`)